### ISSUE TITLE
Introduce support for ChatGPT or ChatGPT API[Feature]

### ISSUE THREAD

**Issue by xiaofeidev** [typora/typora-issues#5655]
ChatGPT is so popular that it is strongly recommended to introduce related support!!!

**Comment by xiaofeidev** [comment]
Typora can directly integrate ChatGPT-related functions through ChatGPT API, or allow users to configure ChatGPT’s API key to take advantage of ChatGPT’s capabilities.

**Comment by yougotwill** [comment]
I think this should be done via a plugin based system sometime in the future. I don't think having ChatGPT built in by default is ideal since it would probably affect performance and require some level of internet connectivity.

**Comment by xiaofeidev** [comment]
> I think this should be done via a plugin based system sometime in the future. I don't think having ChatGPT built in by default is ideal since it would probably affect performance and require some level of internet connectivity.

Agree.

**Comment by yanxiaodi** [comment]
I like Typora but recently I use VS Code more often because it has Copilot - that's amazing. 
It would be great if Typora can enable a plugin for this...

**Comment by Admonstrator** [comment]
Yep, would love to see this as well - but it's really difficult because in normal cases you won't have the memory-feature of ChatGPT, so it would not be too helpfull.

**Comment by jckpn** [comment]
Agree that this would be insanely beneficial to many users of Typora. GitHub Copilot is an amazing example of this -- when typing notes for example, it can autocomplete sentences or fill equations as needed, as well as providing quick explanations while matching the rest of the document's writing style, like here:

![image](https://github.com/typora/typora-issues/assets/14837124/4d7c0a57-c65b-47fb-be26-3c2ee9439a6c)

My current 'workaround' is to use the [OpenAI Playground](https://platform.openai.com/playground) to autocomplete sections for me as needed, but it's pretty cumbersome to have it constantly open in a second window.

**Comment by Admonstrator** [comment]
> My current 'workaround' is to use the [OpenAI Playground](https://platform.openai.com/playground) to autocomplete sections for me as needed, but it's pretty cumbersome to have it constantly open in a second window.

You could use Visual Studio Code paired with GitHub Copilot for it.

**Comment by jiyi27** [comment]
> I like Typora but recently I use VS Code more often because it has Copilot - that's amazing. It would be great if Typora can enable a plugin for this...

Same here!

**Comment by Jiunsr** [comment]
> I like Typora but recently I use VS Code more often because it has Copilot - that's amazing. It would be great if Typora can enable a plugin for this...

I am also looking for related plugins, and to my surprise, someone has already made one. I manually installed [typora-copilot](https://github.com/Snowfly-T/typora-copilot) and it works fine.

**Comment by bucweat** [comment]
MacGPT has an inline feature that uses accessibility to process ChatGPT commands inline in apps. Type the following into a pretty much any app...the `/gpt` is the trigger word to indicate start of your ChatGPT command and [shift+return] is key combo to send string to ChatGPT. The streaming result is entered at the cursor location...here is what that looks like in the apps where this works (vi, TextEdit.app, Notes.app for example):

```
/gpt explain present tense verb conjugation for portuguese verb vestir.[shift+return]
Sure! The verb "vestir" means "to dress" in Portuguese. Here's how it is conjugated in the present tense:
- Eu visto (I dress)
- Tu vestes (You dress - informal)
- Ele/Ela veste (He/She dresses)
- Nós vestimos (We dress)
- Vós vestis (You dress - formal/plural)
- Eles/Elas vestem (They dress)
These conjugations are used when talking about actions happening in the present or general truths. Keep in mind that Portuguese has both formal and informal forms of address, so "tu" and "vós" are used in more informal or regional contexts, while "você" and "vocês" are more common in most situations.
```
In Typora the result is scrambled:
```
/gpt explain present tense verb conjugation for portuguese verb vestir.
Sure! The present tense verbation for conjug the Portuguese verbvestir "" (to dress) is:

 as followsEu visto -
 I dressTu vestes - You dressinformal ()
Ele/Ela veste -/ HeShe dresses
Nós vestimos - We
V dressós vestis - You dressformal (/plural)
Eles/Elas vestem - They

Keep dress in mind thatations verb conjug can vary depending on the context and the subjectoun used. pron
```
It looks like the returned string chunks are inserted out of order. Note that if I switch to source code mode in Typora.app the inline processing no longer works and I'm not sure why that is. My guess is real time parsing/rendering in the WYSIWYG engine is moving the cursor around resulting in somewhat random insertions. 

Maybe someone on the Typora side might be able to explain the scrambling and if there is any solution.
